src/openzaak/components/catalogi/tests/admin/test_published_readonly.py

```diff
@@ -0,0 +1,225 @@
+            verantwoordingsrelatie=["bla"],
+            selectielijst_procestype=procestype_url,
+            doorlooptijd_behandeling=timedelta(days=10),
+            producten_of_diensten=["http://example.com/1", "http://example.com/2"],
+        )
+        url = reverse("admin:catalogi_zaaktype_change", args=(zaaktype.pk,))
+
+        response = self.app.get(url)
+
+        form = response.form
+        form_fields = list(form.fields.keys())
+        zaaktype_fields = [
+            f.name
+            for f in zaaktype._meta.get_fields()
+            if f.name != "datum_einde_geldigheid"
+        ]
+
+        self.assertEqual("datum_einde_geldigheid" in form_fields, True)
+        for field in zaaktype_fields:
+            self.assertEqual(field in form_fields, False)
+
+        # check custom formatting
+        procestype = response.html.find(class_="field-selectielijst_procestype").div.div
+        self.assertEqual(procestype.text, "1 - Instellen en inrichten organisatie")
+        behandeling = response.html.find(
+            class_="field-doorlooptijd_behandeling"
+        ).div.div
+        self.assertEqual(behandeling.text, "10 days")
+        producten_of_diensten = response.html.find(
+            class_="field-producten_of_diensten"
+        ).div.div
+        self.assertEqual(len(producten_of_diensten.find_all("a")), 2)
+
+    def test_readonly_besluittype(self, m):
+        """
+        check that in case of published besluittype only "datum_einde_geldigheid" field is editable
+        """
+        mock_oas_get(m)
+
+        besluittype = BesluitTypeFactory.create(concept=False)
+        url = reverse("admin:catalogi_besluittype_change", args=(besluittype.pk,))
+
+        response = self.app.get(url)
+
+        form = response.form
+        form_fields = list(form.fields.keys())
+        besluittype_fields = [
+            f.name
+            for f in besluittype._meta.get_fields()
+            if f.name != "datum_einde_geldigheid"
+        ]
+
+        self.assertEqual("datum_einde_geldigheid" in form_fields, True)
+        for field in besluittype_fields:
+            self.assertEqual(field in form_fields, False)
+
+    def test_readonly_informatieobjecttype(self, m):
+        """
+        check that in case of published informatieobjecttype only "datum_einde_geldigheid" field is editable
+        """
+        mock_oas_get(m)
+
+        informatieobjecttype = InformatieObjectTypeFactory.create(concept=False)
+        url = reverse(
+            "admin:catalogi_informatieobjecttype_change",
+            args=(informatieobjecttype.pk,),
+        )
+
+        response = self.app.get(url)
+
+        form = response.form
+        form_fields = list(form.fields.keys())
+        informatieobjecttype_fields = [
+            f.name
+            for f in informatieobjecttype._meta.get_fields()
+            if f.name != "datum_einde_geldigheid"
+        ]
+
+        self.assertEqual("datum_einde_geldigheid" in form_fields, True)
```

Comment:
For future reference - you don't have to change it now - there is a `self.assertIn` assertion as well :-)